driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -104,17 +118,23 @@ pub struct OptimisationPriceFinder {
     read_output: fn(&str) -> std::io::Result<String>,
     fee: Option<Fee>,
     optimization_model: OptimizationModel,
+    token_data: TokenData,
 }
 
 impl OptimisationPriceFinder {
-    pub fn new(fee: Option<Fee>, optimization_model: OptimizationModel) -> Self {
+    pub fn new(
+        fee: Option<Fee>,
+        optimization_model: OptimizationModel,
+        token_data: String,
```

Comment:
I think it would be cleaner and more typesafe to already pass the parsed data-structure here instead of an opaque string. We should try to avoid unwrapping in the lower level modules as much as possible.